### DISCUSSION TITLE
Question to adopt langdroid in server-client rest api model.

### DISCUSSION THREAD

**Discussion by 24hours** [langroid/langroid#593] [Q&A]
Hi, I'm planning to adopt langroid and agent framework for our team workflow automation.

The example interaction flow will be as such:
1. User will instruct agent with to perform certain task. Agent is expected to confirm the plan with user ( modification if necessary )
and perform said action. 
2. Example interaction
    - perform code review for PR https://github.com/langroid/langroid/pull/587/files based on our internal coding guideline
    - report our aws servers health status 
    - generate unit test for our code repo based on given error message. 
    - search documentation for certain problem.
3. Each task would be delegated to specialize agent for processing. 
4. Currently, we only plan for chat-based interaction such as https://github.com/open-webui/open-webui and intergration with JIRA / Gitlab / Google worksuite for actual action.

The biggest roadblock I face in this plan. I wish the community will help me clear the doubt.
1. Rest API server such as django-rest-framework ( due to legacy reason ) with langroid as backend logic 
   - chainlit example https://github.com/langroid/langroid/blob/main/examples/chainlit/chat-doc-qa.py#L101 mention `agent.llm_response` however, it doesnt seems to use `Task` which is more commonly how intruction is given in langdroid. 
   - in synchronous environment, it is also unclear how multi-turn interaction should be implemented. This issue also lead into another question.
   
3. Persistent Agent across different session 
   - I have found the FAQ https://langroid.github.io/langroid/FAQ/#can-i-persist-agent-state-across-multiple-runs
   - However, it didnt answer multi-agent / multi task setup.

**Comment by pchalasani** [comment]
You may find this repo useful, as an example of a langroid-based FastAPI server:
https://github.com/langroid/fastapi-server/tree/main

It uses agent , but using task should be similar.

For persistence, doing it with Task (and multi-agents) should be similar.

I am happy to provide more extensive help via a consulting arrangement. Please email me pchalasani@gmail.com if that is of interest.

**Reply by 24hours** [reply]
thanks for the information. I will ask some more question if i think i would help someone else with similar request.
